tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs

```diff
@@ -363,22 +371,25 @@ private async Task AddAllComputeSystemsFromAProvider(ComputeSystemsLoadedData da
     [RelayCommand]
     public void SearchHandler(string query)
     {
-        ComputeSystemCardsView.Filter = system =>
+        var dontShow = new List<int>();
+        for (var i = 0; i < PerProviderViewModels.Count; i++)
         {
-            if (system is CreateComputeSystemOperationViewModel createComputeSystemOperationViewModel)
-            {
-                return createComputeSystemOperationViewModel.EnvironmentName.Contains(query, StringComparison.OrdinalIgnoreCase);
-            }
-
-            if (system is ComputeSystemViewModel computeSystemViewModel)
+            var providerViewModel = PerProviderViewModels[i];
+            providerViewModel.SearchHandler(query);
+            if (!providerViewModel.IsVisible)
             {
-                var systemName = computeSystemViewModel.ComputeSystem!.DisplayName.Value;
-                var systemAltName = computeSystemViewModel.ComputeSystem.SupplementalDisplayName.Value;
-                return systemName.Contains(query, StringComparison.OrdinalIgnoreCase) || systemAltName.Contains(query, StringComparison.OrdinalIgnoreCase);
+                dontShow.Add(i);
             }
+        }
 
-            return false;
-        };
+        // Move all dontShow indexes to the end of the list
+        // so that the visible providers are only shown.
+        foreach (var index in dontShow)
+        {
+            var temp = PerProviderViewModels[index];
+            PerProviderViewModels.RemoveAt(index);
+            PerProviderViewModels.Add(temp);
+        }

```

Comment:
I think there is a bug here if I understand the intent correctly:
Say, there are 3 elements in PerProviderViewModels initially. Let’s call the elements A, B, and C. A and B are not visible. 

Then dontShow will have 0 and 1 indexes in it.

Then the first iteration of this loop will remove A (with index 0) from PerProviderViewModels and add it to the end, so the list will look like B, C, A.
The next iteration of this loop will use index 1 which will remove C from PerProviderViewModels and add it to the end, so the result is B, A, C. And the expected result is C, A, B.

Alternative solution: Use a new collection. Add visible elements first, then not visible. Clear PerProviderViewModels and then add elements from the new collection.

ObservableCollection fires UX update event on every Add/Remove operation which makes UI less responsive. Consider replacing ObservableCollection with BulkObservableCollection and using AddRange or BeginBulkOperation/EndBulkOperation methods to avoid excessive UX events.

Reply:
BulkObservableCollection isn't available in the Nuget list available and is unfortunately only available with a long list of dependent nugets. I have changed the logic to correct it. And I feel that since we won't have a lot of extensions (say 10+), this should be fine. 
But if you feel otherwise, I can work with Eric to add all the nugets needed.

Reply:
That's unfortunate. We should consider it in the future. This is at least the second time I left a comment about excessive events and BulkObservableCollection.

Another idea from this ObservableCollection is to keep them read-only most of the time and use sort order that would include IsVisible property (I didn't research how difficult that would be to implement.